docs/source/model_doc/t5.rst

```diff
@@ -72,34 +88,173 @@ token. T5 can be trained / fine-tuned both in a supervised and unsupervised fash
+  during T5's pre-training.
+
+  However, the example above only shows a single training example. In practice, one trains deep learning models in
+  batches. This entails that we must pad/truncate examples to the same length. For encoder-decoder models, one
+  typically defines a :obj:`max_source_length` and :obj:`max_target_length`, which determine the length of the input
+  and output sequences respectively. For summarization and translation, a good setting is setting
+  :obj:`max_source_length` to 1024 and :obj:`max_target_length` to 128. In addition, we must make sure that padding
+  token id's of the :obj:`labels` are not taken into account by the loss function. In PyTorch, this can be done by
+  replacing them with -100, which is the :obj:`ignore_index` of the :obj:`CrossEntropyLoss`. We also pass
+  :obj:`attention_mask` as additional input to the model, which makes sure that padding tokens of the inputs are
+  ignored. The code example below illustrates all of this.
+
+    .. code-block::
+
+        from transformers import T5ForConditionalGeneration, T5Tokenizer
+        import torch
+
+        model = T5ForConditionalGeneration.from_pretrained("t5-small")
+        tokenizer = T5Tokenizer.from_pretrained("t5-small")
+
+        max_source_length = 1024
+        max_target_length = 128
+
+        # Suppose we have the following 2 training examples:
+        input_sequence_1 = "Welcome to NYC"
+        output_sequence_1 = "Bienvenue à NYC"
+
+        input_sequence_2 = "HuggingFace is a company"
+        output_sequence_2 = "HuggingFace est une entreprise"
+
+        # encode the inputs
+        task_prefix = "translate English to French: "
+        input_sequences = [input_sequence_1, input_sequence_2]
+        encoding = tokenizer([task_prefix + sequence for sequence in input_sequences], padding='max_length', max_length=max_source_length, truncation=True, return_tensors="pt")
+        input_ids, attention_mask = encoding.input_ids, encoding.attention_mask
+
+        # encode the targets
+        target_encoding = tokenizer([output_sequence_1, output_sequence_2], padding='max_length', max_length=max_target_length, truncation=True)
+        labels = target_encoding.input_ids
+
+        # replace padding token id's of the labels by -100
+        labels = [
+                   [(label if label != tokenizer.pad_token_id else -100) for label in labels_example] for labels_example in labels
+        ] 
+        labels = torch.tensor(labels)
+
+        # forward pass
+        loss = model(input_ids=input_ids, attention_mask=attention_mask, labels=labels).loss
+
+Additional training tips:
+
+- T5 models need a slightly higher learning rate than the default one set in the :obj:`Trainer`. Typically, 1e-4 and
+  3e-4 work well for almost all problems (classification, summarization, translation, question answering, question
+  generation).
```

Comment:
Not sure if this is true, cc @patrickvonplaten @patil-suraj.

Also wonder, whether T5 works well with AdamW or whether one should use AdaFactor.

Reply:
In my T5 fine-tuning experiments for most of the tasks using higher lr than the default 5e-5 with Adam worked better. But I'm not sure if this should apply to every task.